android/app/src/main/kotlin/io/lantern/model/SessionModel.kt

```diff
@@ -287,35 +280,56 @@ class SessionModel(
     }
 
     fun updatePaymentMethods(result: MethodChannel.Result?) {
-        lanternClient.plansV3(
-            object : LanternHttpClient.PlansV3Callback {
-                override fun onFailure(
-                    throwable: Throwable?,
-                    error: ProError?,
-                ) {
-                    result?.let {
-                        val errorId = error?.id
-                        activity.runOnUiThread {
-                            result.error("payment_method_fail", errorId, null)
-                        }
-                    }
-                    Logger.error(TAG, "Unable to fetch payment methods: $error", throwable)
-                }
+        val userId = LanternApp.getSession().userId()
+        //Check if not found then call createUserAndFetchPaymentMethods
+        if (userId == 0L) {
+            createUserAndFetchPaymentMethods(result)
```

Comment:
@atavism You were getting that error because the user was not created. This fix should resolve the issue you were encountering. I tested it and it worked fine for me.